### PR TITLE
rpm: build rpm with cmake

### PR DESCRIPTION
**Pull request by tchaikov**
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

**Comment by alimaredia** [comment]
@tchaikov what does the testing for this looking like?

**Comment by tchaikov** [comment]
@alimaredia i haven't run any test suite against this branch. but with the commits in this PR, we are able to build rpm packages with cmake now. i put all commits in this PR except the one which builds rpm using cmake into #10112.

will schedule a rados run later on with the this branch and distro of centos.

**Comment by tchaikov** [comment]
```
2016-07-02T07:38:35.823 INFO:teuthology.orchestra.run.mira101.stderr:Error: Package: 1:ceph-test-11.0.0-231.g1f8e2fc.x86_64 (Ceph)
2016-07-02T07:38:35.823 INFO:teuthology.orchestra.run.mira101.stderr:           Requires: libgmock.so()(64bit)
```

**Comment by tchaikov** [comment]
should not use the "%cmake" macro offered by rpm. it expands to
```
+ /usr/bin/cmake -DCMAKE_C_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_Fortran_FLAGS_RELEASE:STRING=-DNDEBUG -DCMAKE_VERBOSE_MAKEFILE:BOOL=ON -DCMAKE_INSTALL_PREFIX:PATH=/usr -DINCLUDE_INSTALL_DIR:PATH=/usr/include -DLIB_INSTALL_DIR:PATH=/usr/lib64 -DSYSCONF_INSTALL_DIR:PATH=/etc -DSHARE_INSTALL_PREFIX:PATH=/usr/share -DLIB_SUFFIX=64 -DBUILD_SHARED_LIBS:BOOL=ON .. -DCMAKE_INSTALL_LIBEXECDIR=/usr/lib -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_INSTALL_MANDIR=/usr/share/man -DCMAKE_INSTALL_DOCDIR=/usr/share/doc/ceph -DWITH_MANPAGE=ON -DWITH_SYSTEMD=ON -DWITH_CEPHFS_JAVA=ON -DWITH_SELINUX=ON -DWITH_LTTNG=ON -DHAVE_BABELTRACE=ON -DALLOCATOR=tcmalloc
```

in which `-DBUILD_SHARED_LIBS:BOOL=ON` creates shared libraries for gmock.

**Comment by tchaikov** [comment]
tested at

* http://pulpito.ceph.com/kchai-2016-07-02_20:15:06-rados-wip-cmake-rpm---basic-mira
* http://pulpito.ceph.com/kchai-2016-07-04_00:49:16-rados-wip-cmake-rpm---basic-mira/

**Comment by tchaikov** [comment]
changelog

- rebased against master.

**Comment by tchaikov** [comment]
@kdreyer could you take a look also?

File: ceph.spec.in [hunk 1]
```diff
@@ -679,15 +680,19 @@ make %{?_smp_mflags}
 %if 0%{with tests}
 %check
 # run in-tree unittests
-make %{?_smp_mflags} check
+cd build
+ctest -V %{?_smp_mflags}
```
Comment (alimaredia) [inline ceph.spec.in]: this should be `ctest -j` not `ctest -V`. Not sure if we want the '-V' at all.

Comment (tchaikov) [inline ceph.spec.in]: `_smp_mflags` is `-jN`, will drop it then.

**Comment by tchaikov** [comment]
changelog

- dropped "-V" flag for ctest.

**Pull request merged**